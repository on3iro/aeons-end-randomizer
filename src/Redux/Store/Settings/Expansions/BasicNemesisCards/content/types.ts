import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.BasicNemesisCard }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardsContentStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        content: State
      }
    }
  }
}
