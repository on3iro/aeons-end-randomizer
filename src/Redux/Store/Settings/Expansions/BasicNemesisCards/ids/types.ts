import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardIdsStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        ids: string[]
      }
    }
  }
}
