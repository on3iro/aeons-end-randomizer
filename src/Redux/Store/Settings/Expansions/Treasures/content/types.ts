import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.Treasure }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type TreasureContentStateSlice = {
  Settings: {
    Expansions: {
      Treasures: {
        content: State
      }
    }
  }
}
