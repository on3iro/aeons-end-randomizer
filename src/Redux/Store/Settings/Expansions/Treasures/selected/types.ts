import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {
  TOGGLE = 'Settings/Expansions/Treasures/TOGGLE',
  SET_TO_DB = 'Settings/Expansions/Treasures/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/Treasures/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/Treasures/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/Treasures/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/Treasures/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/Treasures/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedTreasuresStateSlice = {
  Settings: {
    Expansions: {
      Treasures: {
        selected: string[]
      }
    }
  }
}
