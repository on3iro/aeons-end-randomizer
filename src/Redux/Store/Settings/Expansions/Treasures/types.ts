import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'types'

import { actions } from './actions'

export type State = Readonly<{
  treasures: SelectedTreasures
  treasureIds: string[]
}>

export enum ActionTypes {
  TOGGLE_CARD = 'Settings/Expansions/Treasures/TOGGLE_CARD',
  SET_TO_DB = 'Settings/Expansions/Treasures/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/Treasures/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/Treasures/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/Treasures/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/Treasures/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/Treasures/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedTreasures = {
  [key: string]: types.Treasure & { selected: boolean }
}

export type TreasuresStateSlice = {
  Settings: {
    Expansions: {
      Treasures: {
        treasures: SelectedTreasures
      }
    }
  }
}

export type TreasureIdsStateSlice = {
  Settings: {
    Expansions: {
      Treasures: {
        treasureIds: string[]
      }
    }
  }
}
