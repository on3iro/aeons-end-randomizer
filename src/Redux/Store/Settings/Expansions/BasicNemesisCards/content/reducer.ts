import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.basicNemesisCards,
  FR: AERData.normalizedData.FR.basicNemesisCards,
  PL: AERData.normalizedData.PL.basicNemesisCards,
  DE: AERData.normalizedData.DE.basicNemesisCards,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
