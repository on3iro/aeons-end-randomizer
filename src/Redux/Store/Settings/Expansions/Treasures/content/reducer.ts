import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.treasures,
  FR: AERData.normalizedData.FR.treasures,
  PL: AERData.normalizedData.PL.treasures,
  DE: AERData.normalizedData.DE.treasures,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
