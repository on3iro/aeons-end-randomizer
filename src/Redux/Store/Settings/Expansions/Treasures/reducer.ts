import { LoopReducer } from 'redux-loop'

import config from 'config'

import { Action, ActionTypes, State } from './types'

import * as reducerHelpers from './reducerHelpers'

export const initialState: State = {
  treasures: config.NORMALIZEDDATA.treasureIds.reduce(
    (acc, id) => ({
      ...acc,
      [id]: {
        ...config.NORMALIZEDDATA.treasures[id],
        selected: true,
      },
    }),
    {}
  ),
  treasureIds: config.NORMALIZEDDATA.treasureIds,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    case ActionTypes.TOGGLE_CARD: {
      return reducerHelpers.toggleCard(state, action)
    }

    case ActionTypes.FETCH_FROM_DB: {
      return reducerHelpers.fetchFromDb(state)
    }

    case ActionTypes.FETCH_FROM_DB_SUCCESS: {
      return reducerHelpers.fetchFromDbSuccess(state, action)
    }

    case ActionTypes.FETCH_FROM_DB_FAILURE: {
      return state
    }

    default: {
      return state
    }
  }
}
