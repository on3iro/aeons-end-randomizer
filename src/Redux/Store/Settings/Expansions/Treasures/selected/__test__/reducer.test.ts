import { Cmd, getCmd, getModel } from 'redux-loop'

import { State } from '../types'
import { actions } from '../actions'
import {
  getSelectedTreasuresFromDB,
  setSelectedTreasuresToDB,
} from '../sideEffects'

import { initialState, Reducer } from '../reducer'

const mockSelectedTreasuresState: State = initialState

describe('Settings | Expansions | Treasures | selected | reducer', () => {
  it('should return the initial state', () => {
    // @ts-ignore
    const result = Reducer(undefined, {})

    expect(result).toEqual(initialState)
  })

  it('should handle TOGGLE', () => {
    const selectedTreasuresToSave = initialState.filter(
      (treasure) => treasure !== 'CronesAmulet'
    )

    const result = Reducer(
      mockSelectedTreasuresState,
      actions.toggle('CronesAmulet')
    )

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toMatchSnapshot()
    expect(cmd).toEqual(
      Cmd.run(setSelectedTreasuresToDB, {
        args: [selectedTreasuresToSave],
        successActionCreator: actions.setToDBSuccessful,
        failActionCreator: actions.setToDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB', () => {
    const result = Reducer(initialState, actions.fetchFromDB())

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toEqual(initialState)

    expect(cmd).toEqual(
      Cmd.run(getSelectedTreasuresFromDB, {
        successActionCreator: actions.fetchFromDBSuccessful,
        failActionCreator: actions.fetchFromDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB_SUCCESS for defined state', () => {
    const result = Reducer(
      mockSelectedTreasuresState,
      actions.fetchFromDBSuccessful(['CronesAmulet'])
    )

    expect(getModel(result)).toEqual(['CronesAmulet'])
  })

  it('should handle FETCH_FROM_DB_SUCCESS for undefined state', () => {
    const result = Reducer(
      initialState,
      // @ts-ignore
      actions.fetchFromDBSuccessful(undefined)
    )

    expect(getModel(result)).toEqual(initialState)
  })

  it.todo('should handle FETCH_FROM_DB_FAILURE')
  it.todo('should handle SET_TO_DB')
  it.todo('should handle SET_TO_DB_SUCCESS')
  it.todo('should handle SET_TO_DB_FAILURE')
})
