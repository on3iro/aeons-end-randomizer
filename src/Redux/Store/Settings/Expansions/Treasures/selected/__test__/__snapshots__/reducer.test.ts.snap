// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings | Expansions | Treasures | selected | reducer should handle TOGGLE 1`] = `
Array [
  "IndirasAetherialGlow",
  "NooksMechanek",
  "ArachnosCobwebs",
  "FossilizedFeather",
  "PortableFurnace",
  "DezmodiasOblivionShard",
  "UlgimorsCoalFragment",
  "IndirasLoneOpal",
  "MistsAmethystParagon",
  "MalastarsImmolate",
  "YanMagdasIlluminate",
  "EyeOfTheMaelstrom",
  "WellOfDespair",
  "EdibleFungusChunks",
  "UmbralHornOfWar",
  "TheBrokenMask",
  "AlchemistsAlembic",
  "SoothsayersPouch",
  "SuppressingPin",
  "Quicksilver",
  "BandOfRetrieval",
  "PurifiedBangle",
  "KadirsEmeraldShard",
  "GexsShatteredGeode",
  "JiansMoonstoneShard",
  "LashsQuartzShard",
  "XaxosFlare",
  "AdelheimsAmethystShard",
  "BramasBuriedLight",
  "GarusTorch",
  "XaxosPyre",
  "MistsGarnetShard",
  "ShimmeringCloakOfTheMagus",
  "ImbuedShackles",
  "GluttonsTooth",
  "CoreOfRage",
  "CleansingAmulet",
  "CarapaceFragement",
  "ThornedWhip",
  "BonesOfDeathmind",
  "AcceleratingGauntlets",
  "EssenceExtractor",
  "BroochOfAttunment",
  "BladedCrystal",
  "PrismOfDestruction",
  "ForgottenTrinket",
  "PrecisionMagnifier",
  "ShroudOfObfucation",
  "VerdantStaff",
  "FlowingMantle",
  "IncosCarnivorousSunflower",
  "GygarsRisingTide",
  "ThraxirsShatteredQuartz",
  "DezmodiasVoidspark",
  "YanMagdasEternalEye",
  "MoltenGold",
  "ClockworkSpindle",
  "BlackenedClaw",
  "RewindPendant",
  "AccelerationStaff",
  "CurrentConductor",
  "VolatileReagent",
  "BeadedCrown",
  "XaxossAetherSiphon",
  "QusOblivionRelic",
  "TalixsEverlight",
  "TalixsEverlight",
  "LostsMoltenForge",
  "AnceintTexts",
  "EmbeddedLight",
  "GarnetDust",
  "WovenMail",
  "SapphireLens",
  "DimensionalPurse",
  "MemoryCoin",
  "InventorsKit",
  "MazahaedronsWorldheartShard",
  "PhaedraxasTourmalineShard",
  "FracturedShell",
  "Extra-DimensionalLens",
  "SeersBracer",
  "RemnantsVoidShard",
  "QuiliussErasure",
  "ZhanasFleetingEmber",
  "BlightedRootClump",
  "PetrifiedWitchFinger",
  "WaywardScraps",
  "TrueSightMonocle",
  "EntwinedAetherStrands",
  "SiphoningBlade",
  "GemEncrustedAnklet",
  "RhiasPlanarPocket",
  "TalixsEverburn",
  "LostsForgedSpark",
  "MazrasThesis",
  "RazrasTrainingWhistle",
  "RejuvenatingDiamond",
  "BrokenTooth",
  "PoisonHeart",
  "WorldMarble",
  "ArmoredPauldrons",
  "BloodshotGeode",
  "TreasuredStone",
  "ReboundGauntlet",
  "KadirsAncientEmerald",
  "NooksNihilum",
  "SmudgedParchment",
  "EmberBlade",
  "WitchsWheel",
  "ClaudiasAetherscope",
  "SoskelsLuckyCoin",
  "QusVoidRadite",
  "TaqrensGiftOfSpirit",
  "GygarsTsunami",
  "EnergizedHusk",
  "SparkingShards",
  "SigilPendant",
  "HornOfPlenty",
  "ImbuedCirclet",
  "BurstingBracelet",
  "IncosSpreadingEmerald",
  "LostsForgedCrystal",
  "OnyxElixir",
  "AshenRib",
  "MassacreHelm",
  "CloakOfWisdom",
  "ReevesRavenShard",
  "NymsVision",
  "SparrowsAid",
  "WraithsEssence",
  "VolatileClasp",
  "BreachStabilizer",
  "FangedChoker",
  "EndlessBandolier",
  "SahalasBlackMeteorite",
  "NooksMysticFlourish",
  "RitualStaff",
  "CrownOfFangs",
  "EverfullPurse",
  "ArcaneTimepiece",
  "BootsOfArcaneQuickening",
  "IlyasTruliteOfForce",
  "KelsTruliteOfEnergy",
  "QusCallOfTheVoid",
  "ThraxirsNamelessSiphon",
  "CairnasInvigorate",
  "DezmodiasEternalDarkness",
  "OhatsShowerOfCoals",
  "LuciensSunShard",
  "RethsFossilizedRoze",
  "TaqrensDefenseGlyphs",
  "ZhanasScimitar",
  "ArachnosSpellbook",
  "IndirasWeatheredOpal",
  "SoskelsTacticalGambit",
  "IlyaAndKelsTourmalineFragment",
  "MalastarsImperectZanite",
  "ClaudiasKaleidoscope",
  "CrackedAntler",
  "SoulEngine",
]
`;
